lib/runtime/storage/trie.go

```diff
@@ -192,18 +211,29 @@ func (t *TrieState) NextKey(key []byte) []byte {
 	defer t.mtx.RUnlock()
 
 	if currentTx := t.getCurrentTransaction(); currentTx != nil {
-		allSortedKeys := append(t.sortedKeys, currentTx.sortedKeys...)
+		mainStateSortedKeys := make([]string, len(t.sortedKeys))
+		copy(mainStateSortedKeys, t.sortedKeys)
+
+		mainStateSortedKeys = slices.DeleteFunc(mainStateSortedKeys, func(s string) bool {
+			_, ok := currentTx.deletes[s]
+			return ok
+		})
+
+		allSortedKeys := append(mainStateSortedKeys, currentTx.sortedKeys...)
+		sort.Strings(allSortedKeys)
```

Comment:
What if instead of creating the slice + copying + deleting + sorting, we just do inserts + sort? I think it could be better in terms of performance

```suggestion
		// Get current tx sorted keys
		allSortedKeys := currentTx.sortedKeys

		// Add main trie sorted keys
		for _, s := range t.sortedKeys {
			if _, ok := currentTx.deletes[s]; !ok {
				allSortedKeys = append(allSortedKeys, s)
			}
		}

		// Sort all
		sort.Strings(allSortedKeys)
```

Reply:
nvm, I benchmarked it and they are almost the same we can keep it as it